src/gui/torrentcontentmodelfolder.cpp

```diff
@@ -32,135 +32,160 @@
-  return m_childItems.count();
+    return m_childItems.count();
 }
 
 // Only non-root folders use this function
 void TorrentContentModelFolder::updatePriority()
 {
-  if (isRootItem())
-    return;
-
-  Q_ASSERT(!m_childItems.isEmpty());
-
-  // If all children have the same priority
-  // then the folder should have the same
-  // priority
-  const int prio = m_childItems.first()->priority();
-  for (int i=1; i<m_childItems.size(); ++i) {
-    if (m_childItems.at(i)->priority() != prio) {
-      setPriority(prio::MIXED);
-      return;
+    if (isRootItem())
+        return;
+
+    Q_ASSERT(!m_childItems.isEmpty());
+
+    // If all children have the same priority
+    // then the folder should have the same
+    // priority
+    const int prio = m_childItems.first()->priority();
+    for (int i = 1; i<m_childItems.size(); ++i) {
+        if (m_childItems.at(i)->priority() != prio) {
+            setPriority(prio::MIXED);
+            return;
+        }
     }
-  }
-  // All child items have the same priority
-  // Update own if necessary
-  setPriority(prio);
+    // All child items have the same priority
+    // Update own if necessary
+    setPriority(prio);
 }
 
 void TorrentContentModelFolder::setPriority(int new_prio, bool update_parent)
 {
-  if (m_priority == new_prio)
-    return;
+    if (m_priority == new_prio)
+        return;
 
-  m_priority = new_prio;
+    m_priority = new_prio;
 
-  // Update parent priority
-  if (update_parent)
-    m_parentItem->updatePriority();
+    // Update parent priority
+    if (update_parent)
+        m_parentItem->updatePriority();
 
-  // Update children
-  if (m_priority != prio::MIXED) {
-    foreach (TorrentContentModelItem* child, m_childItems)
-      child->setPriority(m_priority, false);
-  }
+    // Update children
+    if (m_priority != prio::MIXED)
+        foreach (TorrentContentModelItem* child, m_childItems)
+            child->setPriority(m_priority, false);
 }
 
 void TorrentContentModelFolder::recalculateProgress()
 {
-  qreal tProgress = 0;
-  qulonglong tSize = 0;
-  qulonglong tRemaining = 0;
-  foreach (TorrentContentModelItem* child, m_childItems) {
-    if (child->priority() != prio::IGNORED) {
-      if (child->itemType() == FolderType)
-        static_cast<TorrentContentModelFolder*>(child)->recalculateProgress();
-      tProgress += child->progress() * child->size();
-      tSize += child->size();
-      tRemaining += child->remaining();
+    qreal tProgress = 0;
+    qulonglong tSize = 0;
+    qulonglong tRemaining = 0;
+    foreach (TorrentContentModelItem* child, m_childItems) {
+        if (child->priority() != prio::IGNORED) {
+            if (child->itemType() == FolderType)
+                static_cast<TorrentContentModelFolder*>(child)->recalculateProgress();
+            tProgress += child->progress() * child->size();
+            tSize += child->size();
+            tRemaining += child->remaining();
+        }
+    }
+
+    if (!isRootItem() && ( tSize > 0) ) {
```

Comment:
Really? And I think that is what your code! I didn't see this earlier (in deleted lines).